Stratis.Bitcoin.Features.MemoryPool/MempoolManager.cs

```diff
@@ -15,7 +15,7 @@ namespace Stratis.Bitcoin.Features.MemoryPool
     /// <summary>
     /// A lock for managing asynchronous access to memory pool.
     /// </summary>
-    public class MempoolAsyncLock : AsyncLock
+    public class MempoolAsyncLock : MempoolSchedulerLock
```

Comment:
misunderstanding, I think the origina was like `MempoolAsyncLock : AsyncLock` which was changed to `MempoolAsyncLock : SchedulerLock` and we suggested to name it `MempoolSchedulerLock`, but I believe it was meant to be `MempoolSchedulerLock : SchedulerLock` and not as you changed it. Although I would maybe remove this class completely and just used `SchedulerLock`. 

But if we are about to leave the original design, I believe it should be 
`MempoolSchedulerLock : SchedulerLock`

Reply:
Thanks. Fixed.